### PR TITLE
chore(appeals): upgrading azurerm to version 4

### PR DESCRIPTION
**Pull request by m-juckes-pins**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: infrastructure/service-bus.tf [hunk 1]
```diff
@@ -8,7 +8,6 @@ resource "azurerm_servicebus_namespace" "main" {
 
   sku                          = var.service_bus_config.sku
   capacity                     = var.service_bus_config.capacity
-  zone_redundant               = var.service_bus_config.sku == "Premium"
```
Comment (m-juckes-pins) [inline infrastructure/service-bus.tf:11]: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/eventhub_namespace

Comment (m-juckes-pins) [reply infrastructure/service-bus.tf:11]: possibly need to look somewhere else in terraform to replace this

**Comment by m-juckes-pins** [comment]
Created a pre-commit file but only at the infra level so not to affect anything at root level such as line spacing etc

**Comment by m-juckes-pins** [comment]
- If the provider config in the module is commented out, appeals will pass validation but modules will fail. 
- If the provider block is not commented out appeals will fail validation, but modules will pass, 

``` modules repo
terraform {
  required_providers {
    azurerm = {
      source                = "hashicorp/azurerm"
      version               = ">= 3.0, < 5.0"
      configuration_aliases = [azurerm, azurerm.tooling]
    }
  }
  required_version = ">= 1.5.7, < 1.10.0"
}

# provider "azurerm" {
#   alias = "tooling"
#   features {}
# }
```

**Review by beejjacobs — CHANGES_REQUESTED** [review]
Changes look good. Once the modules change is merged we can update the tag.

**Review by beejjacobs — APPROVED** [review]
Looks good!

**Pull request merged**